Assets/Scripts/DogAI/DogAI.cs

```diff
@@ -0,0 +1,116 @@
+﻿/*
+ * Author(s): Timothy Ng
+ * Description: Brains behind the dogs actions on the world
+ * Usage: [no notes]
+ */
+
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using k = PPGlobal;
+
+public class DogAI : MonoBehaviour {
+
+    enum State
+    {
+        Idle,
+        Wandering,
+        Eating,
+        Pooping
+    }
+
+    State currentState = State.Wandering;
+    Vector2 wanderCenter;
+    float wanderRadius = 500;
+
+    Vector2 target;
+    int frameCounter = 0;
+
+    int tapCount = 0;
+
+	// Use this for initialization
+	void Start () {
+        wanderCenter = GetComponent<RectTransform>().anchoredPosition;
+        target = GetComponent<RectTransform>().anchoredPosition;
+    }
+	
+	void Update () {
+
+        currentState = DecideState();
+
+        switch (currentState)
```

Comment:
Here again, wondering whether we need to run in `Update` or can use coroutines instead which are triggered by `DecideState`. Like to avoid `Update` when possible as it often results in unnecessary calls